### ISSUE TITLE
Icons not used in subgraphs

### ISSUE THREAD

**Issue by x64-latacora** [phronmophobic/clj-graphviz#3]
There's something odd happening with icons when using subgraphs. When creating an edge between a graph and a subgraph, the icon isn't being loaded for the node in the subgraph.

e.g. this:
```clojure
     (let
      [b
       {:edges [["vm" "db"]],
        :nodes [{:id "vm" :image "icons/aws/ec2-instance.png"}
                {:id "db" :image "icons/aws/rds-instance.png"}]
        :default-attributes {:graph {:cluster "true"}
                             :node {:image "icons/question-mark.png"}}}]
       {:edges [["lb" "vm"]],
        :nodes [{:id "lb" :image "icons/aws/elastic-load-balancing.png"}]
        :default-attributes {:graph {:cluster "true"}
                             :node {:image "icons/question-mark.png"}}
        :subgraphs
        [b]})
```

Results in this:

![my-graph](https://github.com/phronmophobic/clj-graphviz/assets/83965621/1bcb75ed-7394-452b-b5cd-489f0cfc49f5)

**Comment by phronmophobic** [comment]
The problem seems to be that if a node is defined in multiple places in the tree, then only the properties of the first definition processed are used.

A work around is to use the same definition in all places. Something like:
```clojure
(let
    [b
     {:edges [["vm" "db"]],
      :nodes [{:id "vm" :image "icons/aws/ec2-instance.png"}
              {:id "db" :image "icons/aws/rds-instance.png"}]
      :default-attributes {:graph {:cluster "true"}
                           :node {:image "icons/question-mark.png"}}}]
  {:edges [["lb" {:id "vm" :image "icons/aws/ec2-instance.png"}]],
   :nodes [{:id "lb" :image "icons/aws/elastic-load-balancing.png"}]
   :default-attributes {:graph {:cluster "true"}
                        :node {:image "icons/question-mark.png"}}
   :subgraphs
   [b]})
```

One possible fix is to:
-  merge all properties every time a node is encountered. 
- multiple nodes with the same id and conflicting property values would be undefined behavior (possibly to become an error in the future).

**Comment by phronmophobic** [comment]
oh whoops. I didn't mean to close yet

**Comment by phronmophobic** [comment]
ok, should be fixed in 0.6.3.
```
com.phronemophobic/clj-graphviz {:mvn/version "0.6.3"}
```

**Comment by x64-latacora** [comment]
Brilliant, thanks @phronmophobic :heart_decoration:
